runtime/near-vm-logic/src/logic.rs

```diff
@@ -2786,6 +2786,67 @@ impl<'a> VMLogic<'a> {
         }
         Ok(())
     }
+
+    /// Verify an ED25519 signature given a message and a public key.
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + ed25519_verify_base + ed25519_verify_byte * num_bytes`
+
+    pub fn ed25519_verify(
+        &mut self,
+        sig_len: u64,
+        sig_ptr: u64,
+        msg_len: u64,
+        msg_ptr: u64,
+        pub_key_len: u64,
+        pub_key_ptr: u64,
+        register_id: u64,
+    ) -> Result<()> {
+        use ed25519_dalek::{PublicKey, Signature, Verifier};
+
+        self.gas_counter.pay_base(ed25519_verify_base)?;
+        self.gas_counter.pay_per(ed25519_verify_byte, msg_len as u64)?;
+        let signature_array = self.get_vec_from_memory_or_register(sig_ptr, sig_len)?;
+        let msg = self.get_vec_from_memory_or_register(msg_ptr, msg_len)?;
+        let pub_key_array = self.get_vec_from_memory_or_register(pub_key_ptr, pub_key_len)?;
+        let pub_key = PublicKey::from_bytes(&pub_key_array)
+            .map_err(|e| VMLogicError::ExternalError(AnyError::new(e.to_string())))?;
+        let signature = Signature::from_bytes(&signature_array)
+            .map_err(|e| VMLogicError::ExternalError(AnyError::new(e.to_string())))?;
+
+        let value = pub_key.verify(&msg, &signature).is_ok();
+        self.internal_write_register(register_id, vec![u8::from(value)])
```

Comment:
as we want to return only a single `u8`, we can just return it directly, there's no need to go through the register.